language/move-binary-format/src/binary_views.rs

```diff
@@ -199,35 +198,158 @@ impl<'a> BinaryIndexedView<'a> {
         }
     }
 
-    pub(crate) fn self_handle_idx(&self) -> Option<ModuleHandleIndex> {
+    pub fn self_handle_idx(&self) -> Option<ModuleHandleIndex> {
         match self {
             BinaryIndexedView::Module(m) => Some(m.self_handle_idx()),
             BinaryIndexedView::Script(_) => None,
         }
     }
 
-    pub(crate) fn module_id_for_handle(&self, module_handle: &ModuleHandle) -> ModuleId {
+    pub fn module_id_for_handle(&self, module_handle: &ModuleHandle) -> ModuleId {
         ModuleId::new(
             *self.address_identifier_at(module_handle.address),
             self.identifier_at(module_handle.name).to_owned(),
         )
     }
 
-    pub(crate) fn self_id(&self) -> Option<ModuleId> {
+    pub fn self_id(&self) -> Option<ModuleId> {
         match self {
             BinaryIndexedView::Module(m) => Some(m.self_id()),
             BinaryIndexedView::Script(_) => None,
         }
     }
 
-    pub(crate) fn version(&self) -> u32 {
+    pub fn version(&self) -> u32 {
         match self {
             BinaryIndexedView::Module(module) => module.version(),
             BinaryIndexedView::Script(script) => script.version(),
         }
     }
 }
 
+// A `BinaryIndexedViewMut` provides table indexed access for `CompiledModuleMut`
+// and `CompiledScriptMut` -- it is the "mutable" counterpart to `BinaryIndexedView`,
+// with one distinction: this class provides access to the underlying collections
+// within the script or module for bounds checking. Accessing collections that cannot exist on
+// a script results in an empty slice, not an error.
+pub enum BinaryIndexedViewMut<'a> {
```

Comment:
We have an issue/task out to kill the mut versions (maybe... probably) At any rate we really shouldn't be building new frameworks on top of them